src/Init/Tactics.lean

```diff
@@ -39,8 +39,74 @@ be a `let` or function type.
 syntax (name := intro) "intro" notFollowedBy("|") (ppSpace colGt term:max)* : tactic
 
 /--
-`intros x...` behaves like `intro x...`, but then keeps introducing (anonymous)
-hypotheses until goal is not of a function type.
+Introduces zero or more hypotheses, optionally naming them.
+
+- `intros` is equivalent to repeatedly applying `intro`
+  until the goal is not an obvious candidate for `intro`.
+  That is, so long as the goal is a `let` or a pi type (e.g. an implication, function, or universal quantifier),
```

Comment:
What do you mean by a `let` here?

Reply:
I mean a `let`, like in `let x := 1; x = x`. There's an example in the examples section.

Reply:
Well, I must say that the example section is very well written and clear, so I have no further comments/suggestions!